src/LoginWithAmazon.test.js

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import TestUtils from 'react';
+import { shallow, mount } from 'enzyme';
+import LoginWithAmazon from './LoginWithAmazon';
+
+let loginWithAmazon;
+let loginWithAmazonInstance;
+let mockSetAccessToken;
+let authorize;
+
+beforeEach(() => {
+    mockSetAccessToken = jest.fn();
+    loginWithAmazon = shallow(<LoginWithAmazon setAccessToken={mockSetAccessToken}/>);
+    loginWithAmazonInstance = loginWithAmazon.instance();
+    authorize = jest.fn();
+
+    Object.defineProperty(window, 'amazon', {
+        value: { Login: { authorize: authorize } },
+        configurable: true
+    });
+});
+
+it('renders without crashing', () => {
+    loginWithAmazon.find('button').simulate('click');
+
+    // Verify authorize called once
+    expect(authorize.mock.calls.length).toBe(1);
+});
+
+it('request access code errors out', () => {
+    global.console = {
+      log: jest.fn()
+    }
+
+    let response={error: 'some_error'}
+    loginWithAmazonInstance.handleResponse(response)
+
+    // Verify error message has been logged to console
+    expect(global.console.log).toHaveBeenCalledWith('Encountered an error on login: some_error')
```

Comment:
May be make the error an object with some fields in it (as against a simple string) to make this test more realistic.

Reply:
?